File: interesting-numbers.cs
using System;

public class Program
{
    public static void Main()
    {    
        while (true) {
            Console.WriteLine("Enter a number: (x to exit)");
            string input = Console.ReadLine();

            if (input == "x") {
                break;    
            } else {
                int n = Convert.ToInt32(input);

                bool isPrime = true;
                bool endsIn5Or3 = false;
                bool digitsAddToTen = false;

                for (int i = n - 1; i > 1; --i) {
                    if (n % i == 0) {
                        isPrime =  false;
                    }
                }

                char[] chars = n.ToString().ToCharArray();
                string[] digits = new string[chars.Length];

                for (int i = 0; i < chars.Length; ++i) {
                    digits[i] = chars[i].ToString();
                }

                int sum = 0;
                for (int i = 0; i < digits.Length; ++i) {
                    sum += Convert.ToInt32(digits[i]);
                }

                if (sum % 10 == 0) {
                    digitsAddToTen = true;
                }

                Console.WriteLine(Convert.ToInt32(digits[digits.Length - 1]));

                if (Convert.ToInt32(digits[digits.Length - 1]) == 3 || Convert.ToInt32(digits[digits.Length - 1]) == 5) {
                    endsIn5Or3 = true;
                }

                Console.WriteLine(isPrime);
                Console.WriteLine(endsIn5Or3);
                Console.WriteLine(digitsAddToTen);

                if (isPrime && endsIn5Or3 && digitsAddToTen) {
                    Console.WriteLine("Number is very interesting");    
                } else if ((isPrime && endsIn5Or3) || (endsIn5Or3 && digitsAddToTen) || (isPrime && digitsAddToTen)) {
                    Console.WriteLine("Number is intersting");    
                } else if (isPrime || endsIn5Or3 || digitsAddToTen) {
                    Console.WriteLine("Number is okay");
                } else {
                    Console.WriteLine("Number is boring");    
                }    
            }
        }    
    }
}

File: interesting-numbers.go
package main

import (
    "fmt"
    "strconv"
)

func main() {
    for {
        fmt.Println("Please enter a number or 'x' to exit:")

        var input string
        fmt.Scanln(&input)

        if input == "x" {
            break
        } else {
            n, _ := strconv.Atoi(input)
            fmt.Println(n)

            isPrime := true
            endsIn5Or3 := false
            digitsAddToTen := false

            for i := n - 1; i > 1; i-- {
                if n % i == 0 {
                    isPrime = false
                }
             }

            nString := strconv.Itoa(n)
            digitsSum := 0

            for _, char := range nString {
                tmp, _ := strconv.Atoi(string(char))
                digitsSum += tmp
            }

            if digitsSum == 10 {
                digitsAddToTen = true;
            }

            if (n % 10) == 5 || (n % 10) == 3 {
                endsIn5Or3 = true
            }

            fmt.Printf("Is the number prime? %t\n", isPrime)
            fmt.Printf("Is the sum of digits 10? %t\n", digitsAddToTen)
            fmt.Printf("Is last digit 3 or 5? %t\n", endsIn5Or3)

            if isPrime == true && endsIn5Or3 == true && digitsAddToTen == true {
                fmt.Printf("he number => %d is very interesting\n", n)
            } else if isPrime == true && endsIn5Or3 == true {
                fmt.Printf("The number => %d is interesting\n", n)
            } else if endsIn5Or3 == true && digitsAddToTen == true {
                fmt.Printf("The number => %d is interesting\n", n)
            } else if isPrime == true && digitsAddToTen == true {
                fmt.Printf("The number => %d is interesting\n", n)
            } else if isPrime == true || endsIn5Or3 == true || digitsAddToTen == true {
                fmt.Printf("The number => %d is almost interesting\n", n)
            } else {
                fmt.Printf("The number => %d is boring\n", n)
            }

            fmt.Println("")
        }
    }
}
